gammapy/cube/exposure.py

```diff
@@ -42,3 +45,59 @@ def make_map_exposure_true_energy(pointing, livetime, aeff, geom):
     exposure = (exposure * livetime).to('m2 s')
 
     return WcsNDMap(geom, exposure.value, unit=exposure.unit)
+
+def make_map_exposure_reco_energy(exposure_map, spectrum, edisp=None):
+    """Create an exposure map in reco energy from an exposure map in true energy.
+
+    Exposure in true energy is weighted with an input spectrum and redistributed in
+    reco energy with the input energy dispersion.
+
+    Parameters
+    ----------
+    exposure_map : `~gammapy.maps.Map`
+        Input exposure map in true energy. unit should have dimension of m2s
+    spectrum : `~gammapy.spectrum.models.SpectralModel`, default is None
+        Spectral model to use to weight exposure in true energy
+        If None is passed, a power law of photon index -2 is assumed.
+    edisp : `~gammapy.irf.EnergyDispersion`, default is None.
+        Energy Dispersion response to redistribute true energies to reco energies
+        If no energy dispersion is passed, a diagonal one is assumed with identical true and reco
+        energy bins. This is the default behaviour.
+
+    Returns
+    -------
+    exposure_image : `~gammapy.maps.Map`
+        Resulting image in reco energy. The unit is the same as the input map.
+    """
+    expo_map = exposure_map.copy()
+    etrue_axis = expo_map.geom.get_axis_by_name("energy")
+    etrue_center = etrue_axis.center * etrue_axis.unit
+    etrue_edges = etrue_axis.edges * etrue_axis.unit
+    binsize = np.diff(etrue_edges)
+
+    if spectrum is None:
+        spectrum = PowerLaw(index=2.0)
+    weights = spectrum(etrue_center)*binsize
+    weights /= weights.sum()
+
+    for img, idx in expo_map.iter_by_image():
+        img *= weights[idx].value
+
+    if edisp is None:
+        edisp = EnergyDispersion.from_diagonal_matrix(etrue_edges, etrue_edges)
```

Comment:
Isn't applying the diagonal matrix a no-op?
If so, you could just now make one and do EDISP stuff if no EDISP is given.
Faster and clearer.